UI/obs-app.cpp

```diff
@@ -1528,9 +1532,42 @@ const char *OBSApp::GetCurrentLog() const
 	return currentLogFile.c_str();
 }
 
+#ifdef __APPLE__
+static DIR *mac_crashlog_directory =
+	opendir(strcat(getenv("HOME"), "/Library/Logs/DiagnosticReports"));
```

Comment:
This is very unsafe as it's initializing a static variable in global scope and also damaging the HOME environment variable:
```
       As typically implemented, getenv() returns a pointer to a string
       within the environment list.  The caller must take care not to
       modify this string, since that would change the environment of
       the process.
```